lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1963,5 +1963,61 @@ sub add_to_temporary_input_files_queue {
     return 1;
 }
 
+sub lock_bam_file_access {
+    my $self = shift;
+
+    unless ($self->get_merged_alignment_results) {
+        my @bams = $self->alignment_bam_file_paths;
+        unless (@bams) {
+            die $self->error_message("Alignment result with class (%s) and id (%s) has neither ".
```

Comment:
It's unlikely considering that longer time will be taken between merge commit  and per lane bam removal. Even it happens and triggers the die, the new build will be ok since merged_alignment_results become available by then.